tests/src/test/scala/akka/kafka/scaladsl/RebalanceExtSpec.scala

```diff
@@ -0,0 +1,427 @@
+
+      // consumer-2::KafkaConsumerActor receives message batch (15,16,17) and forwards it to the **defunct** SubSource-topic-1-1-1-A
+      // consumer-2::SubSource-topic-1-1-1-A:Terminates
+      // consumer-2::Above message sequence is essential for the lost message batch (15,16,17)
+      // consumer-2::DeadLetterListener receives batch (15,16,17) destine for SubSource-topic-1-1-1-A
+
+      // consumer-2::SubSource-topic-1-1-1-B:unblock second message from batch (12,13,14)
+      messageStoreAndAck(13)._2.complete(Try(Done))
+      Await.result(messageStoreAndAck(13)._3.future, maxAwait)
+      // assert(messageStoreAndAck(13)._1.intValue() == 1)
+
+      // consumer-2::SubSource-topic-1-1-0-B:unblock second message from batch (5,6,7)
+      // consumer-2::SubSource-topic-1-1-1-B:issues RequestMessage requesting the next batch
+      messageStoreAndAck(6)._2.complete(Try(Done))
+      Await.result(messageStoreAndAck(6)._3.future, maxAwait)
+      // assert(messageStoreAndAck(6)._1.intValue() == 1)
+
+      // consumer-2::SubSource-topic-1-1-1-B:unblock last message from batch (12,13,14)
+      // consumer-2::SubSource-topic-1-1-1-B:issues RequestMessage requesting the next batch
+      messageStoreAndAck(14)._2.complete(Try(Done))
+      Await.result(messageStoreAndAck(14)._3.future, maxAwait)
+      // assert(messageStoreAndAck(14)._1.intValue() == 1)
+
+      // consumer-2::KafkaConsumerActor receives message batch (18) and forwards it to the SubSource-topic-1-1-1-B
+
+      // unblock all remaining messages
+      val publishedMessageCount = topicCount * partitionCount * perPartitionMessageCount
+      (1 to publishedMessageCount).foreach(
+        messageId =>
+          if (!messageStoreAndAck(messageId)._2.isCompleted) {
+            messageStoreAndAck(messageId)._2.complete(Try(Done))
+          }
+      )
+      // wait until last message from each partition is consumed
+      Await.result(Future.sequence(topicPartitionFutureMap.values.map(_.future)), maxAwait)
+
+      // shutdown system
+      control1.shutdown().futureValue shouldBe Done
+      control2.shutdown().futureValue shouldBe Done
+      sharedKillSwitch1.shutdown()
+      sharedKillSwitch1.shutdown()
+
+      // analyze received messages
+      val consumedMessages = messageStoreAndAck.filter(_._2._1.intValue > 0)
+      log.debug(
+        s"consumedMessages.size=${consumedMessages.size} publishedMessageCount=$publishedMessageCount"
+      )
+      if (consumedMessages.size != publishedMessageCount) {
+        val s1 = 1 to publishedMessageCount
+        val s2 = consumedMessages.keySet
+        log.error(s"FAILURE::missing messages found ${s1.size} != ${s2.size}")
+        s1.filter(!s2.contains(_)).foreach(m => log.error(s"FAILURE::missing message $m topicPartition ${msgTpMap(m)}"))
+      }
+      consumedMessages.size shouldBe publishedMessageCount
+
+      //fail("uncomment me to dump logs for successful run")
```

Comment:
This shouldn't be necessary. If you want logs on a successful run just run this test only and check out `target/kafka.log`.

Reply:
done